src/main/scala/beam/agentsim/agents/household/HouseholdCAVScheduling.scala

```diff
@@ -0,0 +1,396 @@
+package beam.agentsim.agents.household
+import beam.agentsim.agents.planning.BeamPlan
+import org.matsim.api.core.v01.Coord
+import org.matsim.api.core.v01.Id
+
+import org.matsim.api.core.v01.population._
+import org.matsim.core.config.ConfigUtils
+import org.matsim.core.population.PopulationUtils
+import org.matsim.core.utils.geometry.CoordUtils
+
+import scala.collection.immutable.Map
+import scala.collection.immutable.List
+
+sealed trait MobilityServiceRequestType
+case object MSRPickup extends MobilityServiceRequestType
+case object MSRDropoff extends MobilityServiceRequestType
+case object MSRRelocation extends MobilityServiceRequestType
+case object MSRInit extends MobilityServiceRequestType
+
+case class MobilityServiceRequest(
+  person: Option[Person],
+  activity: Activity,
+  time: Double,
+  deltaTime: Double,
+  tag: MobilityServiceRequestType
+) {
+  override def toString =
```

Comment:
Would it be worth wrapping the string with the type, so it can be easily tracked back?

Reply:
Through a trait extending an AnyVal ?

Reply:
No, I had meant something simpler: 
  
    s"MobilityServiceRequest( $tag{ ${person match {
    ...
    )

But I also realized that the values in `$tag` should provided close enough details